### PR TITLE
Add Clair and Security Scanning setup documentation

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: quay-enterprise/clair.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# Clair Setup
+
+<img src="img/Clair_horizontal_color.png" class="img-center" alt="Clair Security Scanner"/>
+
+<a href="https://github.com/coreos/clair">Clair</a> is an open source project for the static analysis
+of vulnerabilities in containers images. It can be used as the engine powering [Quay Security Scanner](security-scanning.md)
```
Comment (joshix) [inline quay-enterprise/clair.md]: s/containers/container/
"The Clair open source code is the engine that powers [Quay Security Scanner]()."

File: quay-enterprise/clair.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+to detect vulnerabilities in all images within Quay, as well as notifiying about new vulnerabilities.
```
Comment (joshix) [inline quay-enterprise/clair.md]: https://github.com/coreos/docs/blob/master/STYLE.md#line-wrapping-considered-harmful

File: quay-enterprise/clair.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+## Initial Setup
+
+### Postgres Database
+
+In order to run Clair, a Postgres database is required. It is recommended to have a Postgres database running
```
Comment (joshix) [inline quay-enterprise/clair.md]: "For production deployments, we recommend a PostgreSQL database running on machines other than those running Quay, and ideally with automatic replication and failover."

File: quay-enterprise/clair.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+on separate machine(s) with automatic failover.
+
+#### Postgres database for testing
+
+For testing only, a postgres database can be started as follows:
```
Comment (joshix) [inline quay-enterprise/clair.md]: "For testing purposes, start a single PostgreSQL instance:"

File: quay-enterprise/clair.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+```
+docker run --name postgres -p 5432:5432 -d postgres
```
Comment (joshix) [inline quay-enterprise/clair.md]: curiosity: Have we tried running this indy postgres container with rkt?

File: quay-enterprise/clair.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+        options:
+          # QUAY_ENDPOINT defines the endpoint at which Quay Enterprise is running.
+          # Example: https://myregistry.mycompany.com
+          registry: {{ QUAY_ENDPOINT }}/keys/
+```
+
+
+### Create Clair configuration (Single-instance)
+
+```
```
Comment (joshix) [inline quay-enterprise/clair.md]:
````markdown
```yaml
```

File: quay-enterprise/clair.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+### Make a configuration directory for Clair
+
+```
+mkdir clair-config
+cd clair-config
+```
+
+## Configure Clair
```
Comment (joshix) [inline quay-enterprise/clair.md]: `###`

File: quay-enterprise/clair.md [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+
+Clair can run either as a single instance or in high-availability mode. It is recommended to run more than a single instance
+of Clair, ideally in an auto-scaling group with automatic healing.
+
+Create a `config.yaml` file in the config directory with the following contents, filling in all places with `{{ VARIABLE }}`:
+
+### Clair configuration (High-availability)
```
Comment (joshix) [inline quay-enterprise/clair.md]: `####`

File: quay-enterprise/clair.md [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+      key_server:
+        type: keyregistry
+        options:
+          # QUAY_ENDPOINT defines the endpoint at which Quay Enterprise is running.
+          # Example: https://myregistry.mycompany.com
+          registry: {{ QUAY_ENDPOINT }}/keys/
+```
+
+
+### Create Clair configuration (Single-instance)
```
Comment (joshix) [inline quay-enterprise/clair.md]: `#### Clair configuration: Single instance`

File: quay-enterprise/clair.md [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+mkdir clair-config
+cd clair-config
+```
+
+## Configure Clair
+
+Clair can run either as a single instance or in high-availability mode. It is recommended to run more than a single instance
+of Clair, ideally in an auto-scaling group with automatic healing.
+
+Create a `config.yaml` file in the config directory with the following contents, filling in all places with `{{ VARIABLE }}`:
```
Comment (joshix) [inline quay-enterprise/clair.md]: s/filling in all places/replacing appearances of {{VARIABLE}} with the appropriate value/

File: quay-enterprise/clair.md [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+```
+
+
+### Configuring Clair for TLS
+
+To configure Clair to run under TLS, a few additional steps are required:
+
+1. Generate a TLS certificate and key pair for the DNS name at which Clair will be accessed
+2. Place these files as `clair.crt` and `clair.key` in your Clair configuration directory
+3. Uncomment the `key_file` and `crt_file` lines under `verifier_proxies` in your Clair `config.json`
```
Comment (joshix) [inline quay-enterprise/clair.md]: config.yaml?

File: quay-enterprise/clair.md (hunk 1 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 2 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Huh!

File: quay-enterprise/clair.md (hunk 3 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 4 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 5 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: I have not

File: quay-enterprise/clair.md (hunk 6 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Added

File: quay-enterprise/clair.md (hunk 8 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 9 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 10 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Changed

File: quay-enterprise/clair.md (hunk 11 above)
Comment (josephschorr) [inline quay-enterprise/clair.md]: Good catch!

**Comment by josephschorr** [comment]
@joshix: Updated

File: quay-enterprise/security-scanning.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
+# Quay Security Scanner
+
+Quay Enterprise supports scanning of container images for known vulnerabilities via a setup security scanner engine such as [Clair](clair.md).
```
Comment (joshix) [inline quay-enterprise/security-scanning.md]: `Quay Enterprise supports scanning container images for known vulnerabilities with a scanning engine such as [Clair](clair.md). This document explains how to configure Clair with Quay Enterprise.`

(Reference links are preferred in new documents, but I can clean these inline links up later.)

File: quay-enterprise/security-scanning.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+<img src="img/security-scanner-endpoint.png" class="img-center" alt="Security Scanner Endpoint"/>
+
+## Generate an auth key
+
+To connect Quay Enterprise securely to the scanner, click "Create Key >" to create an authentication key between Quay and the Security Scanner.
+
+### Authentication for high-availability scanners
+
+If the security scanning engine is running as multiple instances in a high-availability setup, select "Generated shared key":
```
Comment (joshix) [inline quay-enterprise/security-scanning.md]: s/as/on/

File: quay-enterprise/security-scanning.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+<img src="img/security-scanner-generate-shared.png" class="img-center" alt="Security Scanner Generate Shared Key"/>
+
+Enter an optional expiration date, and click "Generate Key":
+
+<img src="img/security-scanner-generate-shared-dialog.png" class="img-center" alt="Security Scanner Generate Shared Key"/>
+
+**Save the key ID and download the preshared private key into the configuration directory for the security scanning engine**
```
Comment (joshix) [inline quay-enterprise/security-scanning.md]: s/engine/engine./

File: quay-enterprise/security-scanning.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+<img src="img/security-scanner-shared-key.png" class="img-center" alt="Security Scanner Shared Key"/>
+
+### Authentication for single-instance scanners
+
+If the security scanning engine is being run as a single instance, select "Have the service provide a key":
```
Comment (joshix) [inline quay-enterprise/security-scanning.md]: s/is being run/is running/

**Comment by joshix** [comment]
@josephschorr Few more nits on the second .md doc

File: quay-enterprise/security-scanning.md (hunk 12 above)
Comment (josephschorr) [inline quay-enterprise/security-scanning.md]: Done

File: quay-enterprise/security-scanning.md (hunk 13 above)
Comment (josephschorr) [inline quay-enterprise/security-scanning.md]: Fixed

File: quay-enterprise/security-scanning.md (hunk 14 above)
Comment (josephschorr) [inline quay-enterprise/security-scanning.md]: Done

File: quay-enterprise/security-scanning.md (hunk 15 above)
Comment (josephschorr) [inline quay-enterprise/security-scanning.md]: Done

**Comment by josephschorr** [comment]
@joshix: Updated

**Comment by joshix** [comment]
LGTM

**Pull request merged**